cpg-core/src/main/kotlin/de/fraunhofer/aisec/cpg/graph/types/HasType.kt

```diff
@@ -134,7 +134,7 @@ interface HasType : ContextProvider, LanguageProvider {
          * A helper function that can be used for [EdgeSingletonList.onChanged]. It unregisters this
          * [TypeObserver] with the [old] node and registers it with the [new] one.
          */
-        fun <NodeType : Node> exchangeTypeObserver(
+        fun <NodeType : Node> exchangeTypeObserverWithAccessPropagation(
```

Comment:
Can't we just add another mandatory (!) parameter `propagateAccess`. Seems error-prone to update two different functions

Reply:
We now have one implementation which we have to maintain but two (external) functions which set the flag. This keeps the code in the nodes a bit smaller.